lib/statsd/instrument/datagram_builder.rb

```diff
@@ -70,20 +72,20 @@ def latency_metric_type
       def normalize_tags(tags)
         return [] unless tags
 
-        tags = tags.map { |k, v| "#{k}:#{v}" } if tags.is_a?(Hash)
+        tags = tags.map { |k, v| "#{k.to_s.gsub(SINGLE_COLON_REGEXP, REPLACEMENT_CHAR)}:#{v.to_s.gsub(SINGLE_COLON_REGEXP, REPLACEMENT_CHAR)}" } if tags.is_a?(Hash)
 
         # Fast path when no string replacement is needed
-        return tags unless tags.any? { |tag| /[|,]/.match?(tag) }
+        return tags unless tags.any? { |tag| /[|, ]/.match?(tag) }
 
-        tags.map { |tag| tag.tr("|,", "") }
+        tags.map { |tag| tag.tr("|, ", REPLACEMENT_CHAR) }
```

Comment:
Are we deliberately using the single whitespace literal here rather than `\s` that would catch a wider variety of whitespaces?

Reply:
Good shout, updated in 196b0cc